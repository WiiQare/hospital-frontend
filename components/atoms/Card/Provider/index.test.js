import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import CardProvider from ".";

describe("CardProvider", () => {
  let component;
  beforeEach(() => {
    const res = render(<CardProvider />);
    component = res.container;
  });

  it("should render the component", () => {
    expect(component).toMatchSnapshot();
  });

  test("should render text elements", () => {
    expect(screen.getByText("Learn more")).toBeInTheDocument();
    expect(screen.getByText("Enroll now")).toBeInTheDocument();
  });
});
