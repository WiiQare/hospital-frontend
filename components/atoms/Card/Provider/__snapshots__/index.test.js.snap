// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardProvider should render the component 1`] = `
<div>
  <div
    class="border hover:shadow rounded-lg py-4 px-5 flex flex-col md:flex-row justify-between gap-6 bg-white cursor-pointer"
  >
    <div
      class="flex gap-3"
    >
      <div
        class="w-32 h-24 md:w-40 md:h-32 border rounded-lg overflow-hidden"
      >
        <img
          alt="bima"
          class="objectif-cover w-full h-full"
          data-nimg="1"
          decoding="async"
          height="24"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="24"
        />
      </div>
      <div
        class="flex flex-col gap-3"
      >
        <div
          class="text-gray-700 space-y-2"
        >
          <h3
            class="text-lg md:text-2xl font-bold"
          >
            B-Stack Family
          </h3>
          <span
            class="text-sm font-light"
          >
            Provided by BIMA
          </span>
        </div>
        <span
          class="flex gap-3 items-center text-sm text-gray-700 font-light"
        >
          <span
            class="bg-orange px-2 py-1 text-white text-sm rounded-md w-fit flex gap-3 items-center"
          >
            3.69 
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              />
            </svg>
          </span>
          17 Plan
        </span>
      </div>
    </div>
    <div
      class="space-y-2"
    >
      <span
        class="py-1 px-2 rounded-full bg-[rgba(254,128,35,.3)] text-xs text-gray-700 font-light flex items-center gap-2 w-fit"
      >
        TeleHealth 
        <svg
          fill="currentColor"
          height="18"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="Circle_Info"
          >
            <g>
              <g>
                <path
                  d="M11.5,15a.5.5,0,0,0,1,0h0V10.981a.5.5,0,0,0-1,0Z"
                />
                <circle
                  cx="12"
                  cy="8.999"
                  r="0.5"
                />
              </g>
              <path
                d="M12,2.065A9.934,9.934,0,1,1,2.066,12,9.945,9.945,0,0,1,12,2.065Zm0,18.867A8.934,8.934,0,1,0,3.066,12,8.944,8.944,0,0,0,12,20.932Z"
              />
            </g>
          </g>
        </svg>
      </span>
      <ul
        class="space-y-2 w-64"
      >
        <li
          class="flex items-center gap-2 text-xs text-gray-700 font-light"
        >
          <svg
            enable-background="new 0 0 48 48"
            fill="currentColor"
            height="18"
            stroke="currentColor"
            stroke-width="0"
            version="1"
            viewBox="0 0 48 48"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              fill="#43A047"
              points="40.6,12.1 17,35.7 7.4,26.1 4.6,29 17,41.3 43.4,14.9"
            />
          </svg>
           Consultation Service 24/7 - Generalists and Specialists
        </li>
        <li
          class="flex items-center gap-2 text-xs text-gray-700 font-light"
        >
          <svg
            enable-background="new 0 0 48 48"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1"
            viewBox="0 0 48 48"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              fill="#43A047"
              points="40.6,12.1 17,35.7 7.4,26.1 4.6,29 17,41.3 43.4,14.9"
            />
          </svg>
           Insurance Benefits GHC 76,000
        </li>
        <li
          class="flex items-center gap-2 text-xs text-gray-700 font-light"
        >
          <svg
            enable-background="new 0 0 48 48"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1"
            viewBox="0 0 48 48"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              fill="#43A047"
              points="40.6,12.1 17,35.7 7.4,26.1 4.6,29 17,41.3 43.4,14.9"
            />
          </svg>
           Medication Benefits GHC 180
        </li>
      </ul>
    </div>
    <div
      class="text-gray-700 flex md:block items-center gap-4 space-y-2"
    >
      <h5
        class="font-light"
      >
        Starts at
      </h5>
      <h3
        class="text-2xl font-semibold"
      >
        $12.15
        <span
          class="font-extralight text-lg"
        >
          /mo
        </span>
      </h3>
      <svg
        aria-hidden="true"
        class="hidden md:block"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13 6a3 3 0 11-6 0 3 3 0 016 0zM18 8a2 2 0 11-4 0 2 2 0 014 0zM14 15a4 4 0 00-8 0v3h8v-3zM6 8a2 2 0 11-4 0 2 2 0 014 0zM16 18v-3a5.972 5.972 0 00-.75-2.906A3.005 3.005 0 0119 15v3h-3zM4.75 12.094A5.973 5.973 0 004 15v3H1v-3a3 3 0 013.75-2.906z"
        />
      </svg>
    </div>
    <div
      class="self-center flex md:flex-col gap-3"
    >
      <button
        class="bg-blue-600 text-white font-extralight py-3 md:px-9 px-5 rounded-lg text-sm"
      >
        Learn more
      </button>
      <button
        class="border border-blue-600 hover:bg-[rgba(28,100,242,.3)] text-blue-600 font-extralight py-3 md:px-9 px-5 rounded-lg text-sm transition duration-300"
      >
        Enroll now
      </button>
    </div>
  </div>
</div>
`;
